streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -183,6 +190,7 @@ private void restorePartition(final Map<TopicPartition, Long> endOffsets,
         final TopicPartition topicPartition = partitionIterator.next();
         final StateRestorer restorer = stateRestorers.get(topicPartition);
         final Long endOffset = endOffsets.get(topicPartition);
+        restorer.maybeNotifyRestoreStarted(restorer.startingOffset(), endOffset);
```

Comment:
I'm thinking that we can simply this logic a bit:

1) in line 124 above, when `needsRestoring.put(topicPartition, restorer);` call `restorer.restoreStarted`.
2) then we can remove the `restoreStarted` boolean in `StoreRestorer` and also the line here.